### PR TITLE
UI external signer support (e.g. hardware wallet)

### PR DESCRIPTION
**Pull request by Sjors**
Big picture overview in [this gist](https://gist.github.com/Sjors/29d06728c685e6182828c1ce9b74483d).

This PR adds GUI support for external signers. It consists of 7 commits on top of #16546 (RPC).

The UX isn't amazing - especially the blocking calls - but it works.

First we adds a GUI setting for the signer script (e.g. path to HWI):

Schermafbeelding 2019-08-05 om 19 32 59

Then we add an external signer checkbox to the wallet creation dialog:

Schermafbeelding 2019-11-07 om 19 17 23

It's checked by default if HWI detects a device. It also grabs the name. It then creates a fresh wallet and imports the keys.

You can verify an address on the device (blocking...):
Schermafbeelding 2019-08-05 om 19 29 22

Sending, including coin selection, Just Works(tm) as long the device is present. It's a bit weird though: the device will prompt for signing first and the QT confirmation dialog appears after.

External signer support is enabled by default when the GUI is configured and Boost::Process is present.

### CODE REVIEW

**Comment by jonasschnelli** [comment]
This PR has 146 commits,... is this intentional?

**Comment by Sjors** [comment]
@jonasschnelli see the list in #16546 for all the all the experimental stuff this PR is based on. The actual change is only a few commits (Github messes up the ordering), compared to RPC-only functionality.

**Comment by Sjors** [comment]
Cleaned it up a bit and tweaked the timestamps so it's more clear that only the most recent 4 commits are relevant.

139 commits may seem a bit intimidating, but it's really just a matter of reviewing:
1. runCommandParseJSON, so the GUI can call HWI: #15382
2. Wallet boxes #16341
3. Native descriptor wallets #16528, which in turn needs a rebase on #16341 and the merged #15450 (create wallet)
4. External signer RPC support: #16546

**Comment by Sjors** [comment]
Rebased. Sending is a bit of hack; I'll clean that up after #18027.

**Comment by Sjors** [comment]
Github lets you mark existing pull requests as draft now!

**Comment by fjahr** [comment]
Concept ACK

I am not a huge GUI user but I compared the way this works here with #16546 and it looks good to me.

**Comment by jonasschnelli** [comment]
I haven't followed the recent HWW work,... but is this still à jour with the concept?

**Comment by Sjors** [comment]
I'm keeping this PR up to date and it actually works, so it *is* the concept :-)

**Comment by Sjors** [comment]
Moving to the new GUI repo: bitcoin-core/gui#4

File: build_msvc/bitcoin_config.h [hunk 1]
```diff
@@ -47,6 +47,12 @@
 /* define if the Boost::Filesystem library is available */
 #define HAVE_BOOST_FILESYSTEM /**/
```
Comment (lindacorbett) [inline build_msvc/bitcoin_config.h:48]: Please do define

**Pull request closed without merging**